lib/liquid/block_body.rb

```diff
@@ -4,8 +4,8 @@
 
 module Liquid
   class BlockBody
-    LiquidTagToken      = /\A\s*(\w+)\s*(.*?)\z/o
-    FullToken           = /\A#{TagStart}#{WhitespaceControl}?(\s*)(\w+)(\s*)(.*?)#{WhitespaceControl}?#{TagEnd}\z/om
+    LiquidTagToken      = /\A\s*([\w#]+)\s*(.*?)\z/o
+    FullToken           = /\A#{TagStart}#{WhitespaceControl}?(\s*)([\w#]+)(\s*)(.*?)#{WhitespaceControl}?#{TagEnd}\z/om
```

Comment:
Because it is a lot harder to revert this decision, do we want to be over protective here?

```suggestion
    LiquidTagToken      = /\A\s*(#?\w+)\s*(.*?)\z/o
    FullToken           = /\A#{TagStart}#{WhitespaceControl}?(\s*)(#?\w+)(\s*)(.*?)#{WhitespaceControl}?#{TagEnd}\z/om
```

Reply:
We don't want to require a word character after the `#`.